dicodile/update_z/distributed_sparse_encoder.py

```diff
@@ -30,13 +32,25 @@ def __init__(self, n_workers, w_world='auto', hostfile=None, verbose=0):
         self.hostfile = hostfile
         self.verbose = verbose
 
-    def init_workers(self, X, D_hat, reg, params, z0=None, DtD=None):
+    def init_workers(self, X, D_hat, reg, params, z0=None,
+                     DtD=None, rank1=False, n_channels=0):  # XXX
+
+        self.rank1 = rank1
+
+        if rank1:
+            assert n_channels > 0, \
+                "n_channels is required to compute rank-1 encoding"
+            uv_hat = D_hat
+            D_hat = get_D(D_hat, n_channels)
```

Comment:
I guess it is fine for a first draft but it would be nice if we can avoid computing this one?

Reply:
OK, we now simply compute the "full" D shape but not D itself